### ISSUE TITLE
error

### ISSUE THREAD

**Issue by ghost** [imbansalaniket/Facial-Recognition-Attandance-System-Using-Python#3]
in get_loc
    return self._engine.get_loc(key)

KeyError: 'Id'

**Comment by imbansalaniket** [comment]
Please elaborate the issue.

**Comment by imbansalaniket** [comment]
> in get_loc
>     return self._engine.get_loc(key)
> 
> KeyError: 'Id'

Please elaborate the issue.

**Comment by ghost** [comment]
It goes ok till Training but after clicking on Track Images, its gives error-

Exception in Tkinter callback
Traceback (most recent call last):
  File "c:\users\user\appdata\local\programs\python\python38\lib\site-packages\pandas\core\indexes\base.py", line 2646, in get_loc
    return self._engine.get_loc(key)
  File "pandas\_libs\index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\hashtable_class_helper.pxi", line 1619, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas\_libs\hashtable_class_helper.pxi", line 1627, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'Id'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\user\appdata\local\programs\python\python38\lib\tkinter\__init__.py", line 1883, in __call__
    return self.func(*args)
  File "<ipython-input-20-bc75ecafb332>", line 206, in TrackImages
    aa=df.loc[df['Id'] == Id]['Name'].values
  File "c:\users\user\appdata\local\programs\python\python38\lib\site-packages\pandas\core\frame.py", line 2800, in __getitem__
    indexer = self.columns.get_loc(key)
  File "c:\users\user\appdata\local\programs\python\python38\lib\site-packages\pandas\core\indexes\base.py", line 2648, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas\_libs\index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\hashtable_class_helper.pxi", line 1619, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas\_libs\hashtable_class_helper.pxi", line 1627, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'Id'


Basically, Not showing Id as Column!

**Comment by imbansalaniket** [comment]
Have you created the blank directories as per the instructions in Readme.md, before training images?

**Comment by ghost** [comment]
Yes, I have created each folder.

**Comment by imbansalaniket** [comment]
Please share the updated StudentDetails.csv file and screenshot of your project folder.

**Comment by ghost** [comment]
![Screenshot (138)](https://user-images.githubusercontent.com/68146613/88160544-672f4580-cc2c-11ea-8dff-0fab5d51459c.png)
![Screenshot (139)](https://user-images.githubusercontent.com/68146613/88160742-b8d7d000-cc2c-11ea-8cbe-1a2e15660eeb.png)

**Comment by imbansalaniket** [comment]
Please add the necessary directories and place files in those directories. It should look like the image below, and the CSV file should contain Id,Name at the first line.
![Screenshot (20)](https://user-images.githubusercontent.com/33326906/88162710-503e2280-cc2f-11ea-88c7-ed8b39d302e7.png)

**Comment by ghost** [comment]
Do we have to write Id and Name manually in CSV?

**Comment by imbansalaniket** [comment]
No, it will automatically happen when you Take Images you have to add attribute name at top example:

Id,Name

1,Aniket

**Comment by ghost** [comment]
Solved!! 
Thank You!!
The problem was with CSV file. I was running program without giving attributes in CSV file. Thank You!

**Comment by imbansalaniket** [comment]
Great! suggest changes if any.
